src/util/Collection.js

```diff
@@ -277,73 +213,80 @@ class Collection extends Map {
+      if (fn(val, key, this)) {
+        results[0].set(key, val);
+      } else {
+        results[1].set(key, val);
+      }
     }
     return results;
   }
 
   /**
-   * Identical to
+   * Maps each item to another value. Identical in behavior to
    * [Array.map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).
    * @param {Function} fn Function that produces an element of the new array, taking three arguments
    * @param {*} [thisArg] Value to use as `this` when executing function
    * @returns {Array}
+   * @example users.map(user => user.tag);
    */
   map(fn, thisArg) {
-    if (thisArg) fn = fn.bind(thisArg);
+    if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
     const arr = new Array(this.size);
     let i = 0;
     for (const [key, val] of this) arr[i++] = fn(val, key, this);
     return arr;
   }
 
   /**
-   * Identical to
+   * Checks if there exists an item that passes a test. Identical in behavior to
    * [Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).
    * @param {Function} fn Function used to test (should return a boolean)
-   * @param {Object} [thisArg] Value to use as `this` when executing function
+   * @param {*} [thisArg] Value to use as `this` when executing function
    * @returns {boolean}
+   * @example users.some(user => user.discriminator === '0000');
    */
   some(fn, thisArg) {
-    if (thisArg) fn = fn.bind(thisArg);
+    if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
     for (const [key, val] of this) {
       if (fn(val, key, this)) return true;
     }
     return false;
   }
 
   /**
-   * Identical to
+   * Checks if all items passes a test. Identical in behavior to
    * [Array.every()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every).
    * @param {Function} fn Function used to test (should return a boolean)
-   * @param {Object} [thisArg] Value to use as `this` when executing function
+   * @param {*} [thisArg] Value to use as `this` when executing function
    * @returns {boolean}
+   * @example users.every(user => !user.bot);
    */
   every(fn, thisArg) {
-    if (thisArg) fn = fn.bind(thisArg);
+    if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
     for (const [key, val] of this) {
       if (!fn(val, key, this)) return false;
     }
     return true;
   }
 
   /**
-   * Identical to
+   * Applies a function to produce a single value. Identical in behavior to
    * [Array.reduce()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce).
    * @param {Function} fn Function used to reduce, taking four arguments; `accumulator`, `currentValue`, `currentKey`,
    * and `collection`
    * @param {*} [initialValue] Starting value for the accumulator
    * @returns {*}
+   * @example guilds.reduce((acc, guild) => acc + guild.memberCount);
```

Comment:
This example needs an initialValue of 0 to work. Else the first time the callback is called, acc is the first Guild object, not its memberCount.